File: samples/ASAdSDKSampleJavaApp/JavaSampleApp/src/main/java/com/appsamurai/adsdk/sample/java/MainActivity.java
package com.appsamurai.adsdk.sample.java;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void gotoProgrammaticBannerAd(View view) {
        gotoActivity(ProgrammaticBannerAdActivity.class);
    }

    public void gotoLayoutBannerAd(View view) {
        gotoActivity(LayoutBannerAdActivity.class);
    }

    public void gotoInterstitialAd(View view) {
        gotoActivity(InterstitialAdActivity.class);
    }

    public void gotoVideoAd(View view) {
        gotoActivity(VideoAdActivity.class);
    }

    private void gotoActivity(Class<?> cls) {
        Intent intent = new Intent(this, cls);
        startActivity(intent);
    }
}

File: samples/ASAdSDKSampleJavaApp/KotlinSampleApp/src/main/java/com/appsamurai/adsdk/sample/kotlin/MainActivity.kt
package com.appsamurai.adsdk.sample.kotlin

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun gotoProgrammaticBannerAd(view: View) {
        gotoActivity(ProgrammaticBannerAdActivity::class.java)
    }

    fun gotoLayoutBannerAd(view: View) {
        gotoActivity(LayoutBannerAdActivity::class.java)
    }

    fun gotoInterstitialAd(view: View) {
        gotoActivity(InterstitialAdActivity::class.java)
    }

    fun gotoVideoAd(view: View) {
        gotoActivity(VideoAdActivity::class.java)
    }

    private fun gotoActivity(cls: Class<*>) {
        val intent = Intent(this, cls)
        startActivity(intent)
    }
}
